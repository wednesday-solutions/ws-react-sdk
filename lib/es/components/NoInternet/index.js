import{_ as e}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"../_rollup-plugin-inject-process-env-0dfb24f4.js";import n from"react";import r from"prop-types";import o from"styled-components";import t from"../T/index.js";import m from"../Warning/index.js";import{c as i}from"../colors-bf7c26a7.js";import{f as a}from"../fonts-56bad7cc.js";import{m as l}from"../media-54d0e07e.js";import"react-intl";import"../If/index.js";var p,s,c,d,f=o.div(p||(p=e(["\n  text-align: center;\n"]))),g=o.img(s||(s=e(["\n  margin-top: 9rem;\n  ","\n  ","\n"])),l.mobile.max("\n    width:75%;\n    margin-top: 12rem;\n  "),l.tablet.max("\n    width:75%;\n    margin-top: 10rem;\n  ")),u=o(t)(c||(c=e(["\n  ",";\n  color: ",";\n  margin: 0rem;\n  ","\n"])),a.style.standard(),i.pawlyBlue,l.largeMobile.max("\n  font-size:0.75rem;\n  ")),x=o(t)(d||(d=e(["\n  ",";\n  color: ",";\n  margin: 0rem 0rem 2rem 0rem;\n  ","\n"])),a.style.standard(),i.pawlyBlue,l.largeMobile.max("\n  font-size:0.75rem;\n  "));function j(e){var{noConnection:r}=e;return n.createElement("div",null,n.createElement(f,null,n.createElement(g,{src:r}),n.createElement(m,{id:"whoof"}),n.createElement(u,{id:"slow_or_no"}),n.createElement(x,{id:"please_check"})))}j.propTypes={noConnection:r.element.isRequired};export default j;export{j as NoInternet};
