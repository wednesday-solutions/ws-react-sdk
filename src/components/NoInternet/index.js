/**
 *
 * NoInternet
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import T from '@components/T';
import Warning from '@components/Warning';
import { colors, fonts, media } from '@themes';

const MainDiv = styled.div`
  text-align: center;
`;
const DogImage = styled.img`
  margin-top: 9rem;
  ${media.mobile.max(`
    width:75%;
    margin-top: 12rem;
  `)}
  ${media.tablet.max(`
    width:75%;
    margin-top: 10rem;
  `)}
`;
const Slow = styled(T)`
  ${fonts.style.standard()};
  color: ${colors.pawlyBlue};
  margin: 0rem;
  ${media.largeMobile.max(`
  font-size:0.75rem;
  `)}
`;
const Check = styled(T)`
  ${fonts.style.standard()};
  color: ${colors.pawlyBlue};
  margin: 0rem 0rem 2rem 0rem;
  ${media.largeMobile.max(`
  font-size:0.75rem;
  `)}
`;
export function NoInternet({ noConnection }) {
  return (
    <div>
      <MainDiv>
        <DogImage src={noConnection} />
        <Warning id="whoof" />
        <Slow id="slow_or_no" />
        <Check id="please_check" />
      </MainDiv>
    </div>
  );
}
NoInternet.propTypes = {
  noConnection: PropTypes.element.isRequired
};
export default NoInternet;
