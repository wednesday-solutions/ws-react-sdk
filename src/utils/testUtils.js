import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
