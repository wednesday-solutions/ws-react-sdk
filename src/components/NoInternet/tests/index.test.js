/**
 *
 * Tests for NoInternet
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import NoInternet from '../index';

describe('<NoInternet />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<NoInternet />);
    expect(baseElement).toMatchSnapshot();
  });
});
