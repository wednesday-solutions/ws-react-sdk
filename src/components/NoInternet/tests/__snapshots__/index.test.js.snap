// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoInternet /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div>
      <div
        class="sc-dlfnbm hWxGDI"
      >
        <img
          class="sc-hKgILt bmiPUu"
        />
        <p
          class="sc-bdfBwQ bZojWc sc-gsTCUz cWOkuF"
          data-testid="warning"
        >
          whoof
        </p>
        <p
          class="sc-bdfBwQ bZojWc sc-eCssSg dfoiiY"
          data-testid="t"
        >
          slow_or_no
        </p>
        <p
          class="sc-bdfBwQ bZojWc sc-jSgupP iCOWkB"
          data-testid="t"
        >
          please_check
        </p>
      </div>
    </div>
  </div>
</body>
`;
